src/main/java/org/spongepowered/api/command/CommandCompletion.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command;
+
+import net.kyori.adventure.text.Component;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.Optional;
+
+/**
+ * A potential completion of a command argument.
+ */
+public interface CommandCompletion {
+
+    /**
+     * Gets a builder that creates a completion.
+     *
+     * @return The {@link Builder}
+     */
+    static Builder builder() {
+        return Sponge.game().builderProvider().provide(Builder.class);
+    }
+
+    /**
+     * Creates a basic completion that only contains {@link String} that a given
+     * input may be completed to.
+     *
+     * @param completion The potential completion
+     * @return The {@link CommandCompletion} object
+     */
+    static CommandCompletion of(final @NonNull String completion) {
```

Comment:
Should have a method that accepts `(String, Component)` too, in my opinion.

Reply:
Use the builder. This is only here because this is going to be by far the most common operation.

Reply:
I don't see how having another convenience method would hurt?

Reply:
imo a builder is excessive given there are only two possible parameters, and no expectation that more will be added in the immediate future. Two methods (String), and (String, Component) would cover the possibilities for a Completion, and make it immediately obvious which combinations of parameters are possible.

Reply:
Fair point. I can get behind that - we can always re-introduce a builder later if it comes to it I guess.